packages/core/src/common/command.ts

```diff
@@ -235,14 +262,20 @@ export class CommandRegistry implements CommandService {
      * Reject if a command cannot be executed.
      */
     // tslint:disable-next-line:no-any
-    async executeCommand<T>(command: string, ...args: any[]): Promise<T | undefined> {
-        const handler = this.getActiveHandler(command, ...args);
+    async executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
+        const handler = this.getActiveHandler(commandId, ...args);
         if (handler) {
             const result = await handler.execute(...args);
+            const command = this.getCommand(commandId);
+            if (command) {
+                const recent = this._recent;
+                this.addRecentCommand(command);
+                this.storageService.setData<{ recent: Command[] }>(this.RECENT_COMMANDS_STORAGE_KEY, { recent });
```

Comment:
please save only once on app stop, @elaihau please help with that, it should be done the same way as for tasks:
- `CommandRegistry` should not be aware about storage but populated from outside
- common frontend contribution should load and save commands state to storage only on start and stop

Reply:
> * common frontend contribution should load and save commands state to storage only on start and stop

we should add the "save and load" logic to this file, should we?

`packages/core/src/browser/common-frontend-contribution.ts`